.github/workflows/build-test.yml

```diff
@@ -58,17 +58,125 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v2
         with:
-          node-version: '14.17.1'
-      - name: restore lerna
-        id: cache
-        uses: actions/cache@v2
-        with:
-          path: |
-            node_modules
-            */*/node_modules
-          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+          node-version: ${{ env.TARGET_NODE_VERSION }}
+          cache: yarn
       - name: Install Dependencies
         run: yarn run bootstrap
+      - uses: actions/cache@v2
+        id: cache-pre-build
+        with:
+          path: ./*
+          key: pre-build-${{ github.sha }}
+  docker-artifacts:
+    name: Build airnode-artifacts and push to Docker Hub
+    runs-on: ubuntu-latest
+    needs: pre-build
+    steps:
+      - uses: actions/cache@v2
+        id: cache-pre-build
+        with:
+          path: ./*
+          key: pre-build-${{ github.sha }}
+      - name: Build airnode-artifacts
+        run: |
+          docker build --tag api3/airnode-artifacts-dev:${{ github.sha }} --file docker/Dockerfile .
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Push airnode artifacts to Docker Hub
+        run: |
+          docker push api3/airnode-artifacts-dev:${{ github.sha }}
+  docker-airnode-admin:
```

Comment:
Since the docker admin, deployer, and client build steps all take about the same time and have very similar code, what about a matrix strategy for improving legibility?

Reply:
Sure :)

Reply:
I've implemented this change, but I had to rename the one docker build command in the root `package.json`.

The last build took 11 m 28 s, which is about what it was before this PR :smile: